### PR TITLE
Upgrading google-cloud-bom and its associated shared dependencies

### PR DESCRIPTION
**Pull request by suztomo**
Upgrading google-cloud-bom 0.159.0 and updating other dependencies referencing the shared dependencies BOM 2.0.1 (as per Neenu) (https://search.maven.org/artifact/com.google.cloud/google-cloud-shared-dependencies/2.0.1/pom).

- gax-bom 2.1.0 has https://search.maven.org/artifact/com.google.api/gax-bom/2.1.0/pom

```diff
diff --git a/pom.xml b/pom.xml
index 3f58b57..0685cdd 100644
--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
   <groupId>com.google.cloud</groupId>
   <artifactId>google-cloud-shared-dependencies</artifactId>
   <packaging>pom</packaging>
-  <version>1.4.0</version><!-- {x-version-update:google-cloud-shared-dependencies:current} -->
+  <version>2.0.1</version><!-- {x-version-update:google-cloud-shared-dependencies:current} -->
   <name>Google Cloud Shared Dependencies</name>
   <url>https://github.com/googleapis/java-shared-dependencies</url>
   <description>
@@ -14,7 +14,7 @@
   <parent>
     <groupId>com.google.cloud</groupId>
     <artifactId>google-cloud-shared-config</artifactId>
-    <version>0.12.0</version>
+    <version>1.0.0</version>
   </parent>
 
   <organization>
@@ -36,11 +36,11 @@
   <distributionManagement>
     <snapshotRepository>
       <id>sonatype-nexus-snapshots</id>
-      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+      <url>https://google.oss.sonatype.org/content/repositories/snapshots</url>
     </snapshotRepository>
     <repository>
       <id>sonatype-nexus-staging</id>
-      <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
+      <url>https://google.oss.sonatype.org/service/local/staging/deploy/maven2/</url>
     </repository>
   </distributionManagement>
 
@@ -56,14 +56,14 @@
     <site.installationModule>${project.artifactId}</site.installationModule>
 
     <grpc.version>1.39.0</grpc.version>
-    <gax.version>1.66.0</gax.version>
-    <grpc-gcp.version>1.0.0</grpc-gcp.version>
-    <guava.version>30.1.1-android</guava.version>
+    <gax.version>2.1.0</gax.version>
+    <grpc-gcp.version>1.1.0</grpc-gcp.version>
+    <guava.version>30.1.1-jre</guava.version>
     <protobuf.version>3.17.3</protobuf.version>
-    <google.api-common.version>1.10.4</google.api-common.version>
+    <google.api-common.version>2.0.1</google.api-common.version>
     <google.common-protos.version>2.3.2</google.common-protos.version>
-    <google.core.version>1.95.4</google.core.version>
-    <google.auth.version>0.26.0</google.auth.version>
+    <google.core.version>2.0.5</google.core.version>
+    <google.auth.version>1.0.0</google.auth.version>
     <google.http-client.version>1.39.2</google.http-client.version>
     <google.oauth-client.version>1.31.5</google.oauth-client.version>
     <google.api-client.version>1.32.1</google.api-client.version>
@@ -74,8 +74,8 @@
     <iam.version>1.0.14</iam.version>
     <opencensus.version>0.28.0</opencensus.version>
     <findbugs.version>3.0.2</findbugs.version>
-    <errorprone.version>2.7.1</errorprone.version>
-    <jackson.version>2.12.3</jackson.version>
+    <errorprone.version>2.8.1</errorprone.version>
+    <jackson.version>2.12.4</jackson.version>
     <codec.version>1.15</codec.version>
     <httpcomponents.httpcore.version>4.4.14</httpcomponents.httpcore.version>
     <httpcomponents.httpclient.version>4.5.13</httpcomponents.httpclient.version>
```

### CODE REVIEW

**Comment by suztomo** [comment]
```
Require upper bound dependencies error for com.google.guava:guava:30.1.1-android paths to dependency are:
+-com.google.cloud.tools.opensource:upper-bounds-check:2.0.0-SNAPSHOT
  +-com.google.cloud:google-cloud-storage:2.0.1
    +-com.google.guava:guava:30.1.1-android (managed) <-- com.google.guava:guava:30.1.1-jre
and
+-com.google.cloud.tools.opensource:upper-bounds-check:2.0.0-SNAPSHOT
  +-com.google.cloud:google-cloud-core:2.0.5
    +-com.google.guava:guava:30.1.1-android (managed) <-- com.google.guava:guava:30.1.1-jre
, 
Require upper bound dependencies error for com.google.api:api-common:1.10.4 paths to dependency are:
+-com.google.cloud.tools.opensource:upper-bounds-check:2.0.0-SNAPSHOT
  +-com.google.cloud:google-cloud-storage:2.0.1
    +-com.google.api:api-common:1.10.4 (managed) <-- com.google.api:api-common:2.0.1
and
+-com.google.cloud.tools.opensource:upper-bounds-check:2.0.0-SNAPSHOT
  +-com.google.cloud:google-cloud-core:2.0.5
    +-com.google.api:api-common:1.10.4 (managed) <-- com.google.api:api-common:2.0.1
]
```

File: boms/cloud-oss-bom/pom.xml [hunk 1]
```diff
@@ -44,17 +44,18 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <guava.version>30.1.1-android</guava.version>
-    <google.cloud.java.version>0.158.0</google.cloud.java.version>
-    <google.cloud.core.version>1.95.4</google.cloud.core.version>
+    <guava.version>30.1.1-jre</guava.version>
```
Comment (suztomo) [inline boms/cloud-oss-bom/pom.xml:47]: From `<guava.version>30.1.1-jre</guava.version>` in https://search.maven.org/artifact/com.google.cloud/google-cloud-shared-dependencies/2.0.1/pom

File: boms/cloud-oss-bom/pom.xml [hunk 2]
```diff
@@ -44,17 +44,18 @@
...
+    <google.cloud.java.version>0.159.0</google.cloud.java.version>
+    <google.cloud.core.version>2.0.5</google.cloud.core.version>
```
Comment (suztomo) [inline boms/cloud-oss-bom/pom.xml:49]: From `<google.core.version>2.0.5</google.core.version>` in https://search.maven.org/artifact/com.google.cloud/google-cloud-shared-dependencies/2.0.1/pom

File: boms/cloud-oss-bom/pom.xml [hunk 3]
```diff
@@ -44,17 +44,18 @@
...
     <io.grpc.version>1.39.0</io.grpc.version>
     <http.version>1.39.2</http.version>
     <protobuf.version>3.17.3</protobuf.version>
     <!-- We don't use gax-bom because it includes the artifacts with 'testlib' classifier.
         When updating gax.version, update gax.httpjson.version too. -->
-    <gax.version>1.66.0</gax.version>
-    <gax.httpjson.version>0.83.0</gax.httpjson.version>
-    <auth.version>0.26.0</auth.version>
+    <gax.version>2.1.0</gax.version>
+    <gax.httpjson.version>0.86.0</gax.httpjson.version>
```
Comment (suztomo) [inline boms/cloud-oss-bom/pom.xml:56]: From `<gax.version>2.1.0</gax.version>` in https://search.maven.org/artifact/com.google.cloud/google-cloud-shared-dependencies/2.0.1/pom.

gax-bom 2.1.0 has https://search.maven.org/artifact/com.google.api/gax-bom/2.1.0/pom, where gax-httpjson 0.86.0

File: boms/cloud-oss-bom/pom.xml [hunk 4]
```diff
@@ -44,17 +44,18 @@
...
+    <auth.version>1.0.0</auth.version>
```
Comment (suztomo) [inline boms/cloud-oss-bom/pom.xml:57]: From `<google.auth.version>1.0.0</google.auth.version>` in https://search.maven.org/artifact/com.google.cloud/google-cloud-shared-dependencies/2.0.1/pom

File: boms/cloud-oss-bom/pom.xml [hunk 5]
```diff
@@ -44,17 +44,18 @@
...
+    <api-common.version>2.0.1</api-common.version>
```
Comment (suztomo) [inline boms/cloud-oss-bom/pom.xml:58]: From `<google.api-common.version>2.0.1</google.api-common.version>` in https://search.maven.org/artifact/com.google.cloud/google-cloud-shared-dependencies/2.0.1/pom

File: boms/cloud-oss-bom/pom.xml [hunk 6]
```diff
@@ -69,12 +70,16 @@
 
   <dependencyManagement>
     <dependencies>
+      <!-- Not using Guava-bom because it includes guava-gwt, which has many invalid references -->
```
Comment (suztomo) [inline boms/cloud-oss-bom/pom.xml:73]: guava-gwt is not included in guava-bom 30.1.1-android

File: boms/cloud-oss-bom/pom.xml (hunk 1 above)
Comment (chanseokoh) [reply boms/cloud-oss-bom/pom.xml:47]: Interesting. The reason I heard that we were using `-android` is to support Java 7. Does this mean that we can forget Java 7 from now on?

**Review by dzou — APPROVED** [review]
LGTM. Perhaps it may be helpful to add a comment for each set of dependency versions which bom those versions are coming from (if I understand correctly, we must maintain certain versions of dependencies manually since including their bom may yield conflicts).

**Comment by suztomo** [comment]
Except the google-cloud-bom version, they are all coming from the shared dependencies BOM.

**Comment by dzou** [comment]
> Except the google-cloud-bom version, they are all coming from the shared dependencies BOM.

Ah got it. Thanks.

**Pull request merged**